src/components/ProductGrid.tsx

```diff
@@ -29,29 +29,31 @@ const ProductGrid = () => {
 
   console.log(rankType, targetType)
 
-  const baseUrl = import.meta.env.VITE_BASE_URL
-
-  const handleGoOrder = useCallback(
+  const handleGoProduct = useCallback(
     (id: number) => {
       if (!isLoggedIn) {
         navigate(ROUTES.LOGIN)
       } else {
-        navigate(getRoute(ROUTES.ORDER, { id }))
+        navigate(getRoute(ROUTES.PRODUCT, { id }))
       }
     },
     [isLoggedIn, navigate]
   )
 
-  const rankingUrlObj = new URL("/api/products/ranking", baseUrl)
-  rankingUrlObj.searchParams.set("targetType", targetType)
-  rankingUrlObj.searchParams.set("rankType", rankType)
-  const rankingUrl = rankingUrlObj.toString()
-
-  const { data: productsData, isLoading: loading } = useQuery<ProductsResponse>({
-    queryKey: ["products", "ranking", rankType, targetType],
-    queryFn: () => axios.get<ProductsResponse>(rankingUrl).then((res) => res.data),
-    enabled: !!rankType && !!targetType,
-  })
+  const { data: productsData, isLoading: loading } = useQuery<ProductsResponse>(
+    {
+      queryKey: ["products", "ranking", rankType, targetType],
+      queryFn: () => {
+        const params = new URLSearchParams()
+        params.append("targetType", targetType)
+        params.append("rankType", rankType)
+        return axiosInstance
+          .get<ProductsResponse>(`/api/products/ranking?${params.toString()}`)
+          .then((res) => res.data)
```

Comment:
axios의 params 옵션을 사용하면 URLSearchParams를 사용하지 않아도 아래와 같이 간단하게 표현할 수 있습니다.

```suggestion
        return axiosInstance
          .get<ProductsResponse>(`/api/products/ranking`, {
            params: {
               targetType,
               rankType
            }
          })
          .then((res) => res.data)
```

Reply:
아 저번 피드백에 말씀해주셨던 부분인데 놓쳤네요 ㅜ 감사합니다!